crypto/sr25519/pubkey.go

```diff
@@ -0,0 +1,71 @@
+package sr25519
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+
+	schnorrkel "github.com/ChainSafe/go-schnorrkel"
+)
+
+var _ crypto.PubKey = PubKeySr25519{}
+
+// PubKeySr25519Size is the number of bytes in an Sr25519 signature.
+const PubKeySr25519Size = 32
+
+// PubKeySr25519 implements crypto.PubKey for the Sr25519 signature scheme.
+type PubKeySr25519 [PubKeySr25519Size]byte
+
+// Address is the SHA256-20 of the raw pubkey bytes.
+func (pubKey PubKeySr25519) Address() crypto.Address {
+	return crypto.Address(tmhash.SumTruncated(pubKey[:]))
+}
+
+// Bytes marshals the PubKey using amino encoding.
+func (pubKey PubKeySr25519) Bytes() []byte {
+	bz, err := cdc.MarshalBinaryBare(pubKey)
+	if err != nil {
+		panic(err)
+	}
+	return bz
+}
+
+func (pubKey PubKeySr25519) VerifyBytes(msg []byte, sig []byte) bool {
+	// make sure we use the same algorithm to sign
+	if len(sig) != SignatureSize {
+		return false
+	}
+	var sig64 [SignatureSize]byte
+	copy(sig64[:], sig)
+
+	publicKey := &(schnorrkel.PublicKey{})
+	err := publicKey.Decode(pubKey)
+	if err != nil {
+		return false
+	}
+
+	signingContext := schnorrkel.NewSigningContext([]byte{}, msg)
+
+	signature := &(schnorrkel.Signature{})
+	err = signature.Decode(sig64)
+	if err != nil {
+		return false
+	}
+
+	return publicKey.Verify(signature, signingContext)
+}
+
+func (pubKey PubKeySr25519) String() string {
+	return fmt.Sprintf("PubKeySr25519{%X}", pubKey[:])
+}
+
+// nolint: golint
+func (pubKey PubKeySr25519) Equals(other crypto.PubKey) bool {
+	if otherEd, ok := other.(PubKeySr25519); ok {
+		return bytes.Equal(pubKey[:], otherEd[:])
```

Comment:
not a constant time compare?

Reply:
Idk, this was the way we did it in the Ed25519 version.  https://github.com/tendermint/tendermint/blob/master/crypto/ed25519/ed25519.go#L164

Reply:
probably not needed